Library/OptimizelyClientType.swift

```diff
@@ -62,10 +63,63 @@ extension OptimizelyClientType {
 
     return variant
   }
+
+  /* Returns all experiments the app knows about */
+
+  public func allExperiments() -> [OptimizelyExperiment.Key] {
+    return OptimizelyExperiment.Key.allCases
+  }
 }
 
 // MARK: - Tracking Properties
 
+public func optimizelyProperties(environment: Environment? = AppEnvironment.current) -> [String: Any]? {
+  guard let env = environment, let optimizelyClient = env.optimizelyClient else {
+    return nil
+  }
+
+  let environmentType = env.environmentType
+  let userId = deviceIdentifier(uuid: UUID())
+  let attributes = optimizelyUserAttributes()
```

Comment:
I wrote a whole question but Github ate it?

I was asking if we always call this with no project and ref tags, those properties will always be `nil`. Since  `optimizelyUserAttributes ` calls `sessionRefTagProperties` on line 159. Do those get set somewhere else?

Reply:
I thought about this and decided that it doesn't make any sense that we have `ref_tag` and `referrer_credit` as user properties 😅 I don't see a world where we would use those properties to modify the audience of an experiment, and for that reason I decided it was simpler to always leave them out. Will check with product whether this is ok though!